src/app/ui/editor/moving_pixels_state.cpp

```diff
@@ -285,6 +286,20 @@ bool MovingPixelsState::onMouseDown(Editor* editor, MouseMessage* msg)
   Decorator* decorator = static_cast<Decorator*>(editor->decorator());
   Doc* document = editor->document();
 
+  // Return if it cannot draw in current layer
+  TransformHandles* transfHandles = decorator->getTransformHandles(editor);
+  HandleType handle = transfHandles->getHandleAtPoint(editor,
+                                                      msg->position(),
+                                                      getTransformation(editor));
+  if (editor->isInsideSelection() || handle != NoHandle) {
+    if (layer_is_locked(editor))
+      return true;
+  }
+  else {
+    dropPixels();
+    return StandbyState::onMouseDown(editor, msg);
+  }
+
```

Comment:
What specific case is this filtering? Isn't this related to the condition below? Couldn't we add a `layer_is_locked` check just before `m_pixelsMovement->catchImageAgain(...)` below?  (not sure about this because I don't fully understand why this code is here now)